vaadin-date-picker-flow-parent/vaadin-date-picker-flow/src/main/java/com/vaadin/flow/component/datepicker/DatePicker.java

```diff
@@ -384,6 +394,19 @@ private void setI18nWithJS() {
             getElement().executeJs(
                     "this.i18n = Object.assign({}, this.i18n, $0);",
                     i18nObject);
+
+            boolean hasDateFormats = i18n.dateFormats != null
+                    && i18n.dateFormats.size() > 0;
+
+            if (hasDateFormats) {
```

Comment:
Similarly we can add a comment here to explain the branching.

Reply:
put the same comment as above here